shared-libs/transitions/src/transitions/registration.js

```diff
@@ -185,6 +186,27 @@ const setBirthDate = doc => {
   }
 };
 
+const getPhoneNumber = doc => {
+  if (!doc || !doc.fields) {
+    return '';
+  }
+  return doc.fields.phone_number;
+};
+
+const setPhoneNumber = doc => {
+  const phoneNumber = getPhoneNumber(doc);
+  // By default for a valid phone number SmsParser itself adds a country code if not provided
+  // So if country code is not present in phone_number throw error
+  const validPhone = phoneNumberParser.validate(config.getAll(), phoneNumber);
+  if (!validPhone) {
+    throw new Error(
+      `${doc.phoneNumber} submitted by ${doc.from} is not a valid phone number`
```

Comment:
Can we route this to an event and allow a custom message to be sent similar to the `patient_not_found` event?
https://github.com/medic/cht-core/blob/1d19531b2404595d17f63b631c659a717a786ee5/tests/integration/sentinel/transitions/update-notifications.spec.js#L116

Reply:
Also, what if we don't want to allow duplicate phone numbers? What if others want to allow it? Can that be made configurable?

Reply:
It seems we can. Thanks. I added a phone not valid event and its translations (en , ne).